air-quality-ui/system_tests/ui/summary_page/summary_page_table_data.spec.ts

```diff
@@ -1,487 +1,813 @@
+      pm2_5: {
+        mean: { aqi_level: CaseAQI6.aqiLevel, value: CaseAQI6.pm2_5 },
+      },
+      pm10: { mean: { aqi_level: CaseAQI2.aqiLevel, value: CaseAQI2.pm10 } },
+      so2: { mean: { aqi_level: CaseAQI1.aqiLevel, value: CaseAQI1.so2 } },
+    }),
+    createMeasurementSummaryAPIResponseData({
+      measurement_base_time: '2024-06-19T12:00:00Z',
+      location_name: 'Abu Dhabi',
+      overall_aqi_level: { mean: CaseAQI4.aqiLevel },
+      no2: { mean: { aqi_level: CaseAQI1.aqiLevel, value: CaseAQI1.no2 } },
+      o3: { mean: { aqi_level: CaseAQI5.aqiLevel, value: CaseAQI5.o3 } },
+      pm2_5: {
+        mean: { aqi_level: CaseAQI4.aqiLevel, value: CaseAQI4.pm2_5 },
+      },
+      pm10: { mean: { aqi_level: CaseAQI4.aqiLevel, value: CaseAQI4.pm10 } },
+      so2: { mean: { aqi_level: CaseAQI1.aqiLevel, value: CaseAQI1.so2 } },
+    }),
+    createMeasurementSummaryAPIResponseData({
+      measurement_base_time: '2024-06-19T12:00:00Z',
+      location_name: 'Zurich',
+      overall_aqi_level: { mean: CaseAQI3.aqiLevel },
+      no2: {
+        mean: { aqi_level: CaseAQI1.aqiLevel, value: CaseAQI1.no2 },
+      },
+      o3: { mean: { aqi_level: CaseAQI3.aqiLevel, value: CaseAQI3.o3 } },
+      pm2_5: {
+        mean: { aqi_level: CaseAQI2.aqiLevel, value: CaseAQI2.pm2_5 },
+      },
+      pm10: { mean: { aqi_level: CaseAQI2.aqiLevel, value: CaseAQI2.pm10 } },
+      so2: { mean: { aqi_level: CaseAQI1.aqiLevel, value: CaseAQI1.so2 } },
+    }),
+  ]
+  await summaryPage.setupPageWithMockData(
+    mockedForecastResponse,
+    mockedMeasurementSummaryResponse,
   )
-  await summaryPage.getColumnHeaderAndText('Ozone (µg/m³)', 'Ozone (µg/m³)')
-  await summaryPage.getColumnHeaderAndText(
-    'Sulphur Dioxide (µg/m³)',
-    'Sulphur Dioxide (µg/m³)',
+  const expectedData = [
+    // Kampala
+    [
+      // AQI Level
+      CaseAQI2.aqiLevel.toString(), // Forecast
+      CaseAQI6.aqiLevel.toString(), // Measured
+      '-4', // Diff
+      // pm2.5
+      CaseAQI2.pm2_5.toString(), // Forecast
+      CaseAQI6.pm2_5.toString(), // Measured
+      '19 Jun 09:00', // Time
+    ],
+    // Abu Dhabi
+    [
+      // AQI Level
+      CaseAQI5.aqiLevel.toString(), //Forecast
+      CaseAQI4.aqiLevel.toString(), //Measured
+      '+1', // Diff
+      //pm2.5
+      CaseAQI5.pm2_5.toString(), //Forecast
+      CaseAQI4.pm2_5.toString(), // Measured
+      '19 Jun 12:00', //Time
+    ],
+    // Zurich
+    [
+      // AQI Level
+      CaseAQI3.aqiLevel.toString(), //Forecast
+      CaseAQI3.aqiLevel.toString(), //Measured
+      '0', // Diff
+      //pm2.5
+      CaseAQI2.pm2_5.toString(), //Forecast
+      CaseAQI2.pm2_5.toString(), //Measured
+      '19 Jun 12:00', //Time
+    ],
+  ]
+
+  await summaryPage.assertGridAttributes('values', expectedData)
+})
+
+test('(BugFix #191) Verify pollutant level diff 0 does not override a larger diff for calculation of overall AQI level', async ({
```

Comment:
Do we need to specifiy this test is because of a bug?

Reply:
Yes this went in already actually on ticket #191, I think it's showing green as I just moved it's place in the file

Reply:
I'm not a fan of listing bug numbers/ ticket numbers in code in general. It's not relevant to the test and in my view is just a bit of clutter. It's good that the test is here, and that the test gap is now filled, just not a fan having the test number in the test name..

Reply:
Ok no problem, removed